### DISCUSSION TITLE
New Coinbase Python SDK

### DISCUSSION THREAD

**Discussion by jonandtice** [KmiQ/coinbase-advanced-python#47] [General]
I'm sure @KmiQ you've seen that Coinbase has released their own [Python SDK](https://docs.cloud.coinbase.com/advanced-trade-api/docs/sdk-overview) that is pretty much the same as what you are aiming to do here.

I was just wondering how the official SDK affects the plans for this codebase. I hope to continue using this one since that's what I already started using and I prefer how this one creates objects instead of just dumping the json. So I hope the official SDK eases development of this one rather than replacing it.

**Comment by KmiQ** [comment]
> I'm sure @KmiQ you've seen that Coinbase has released their own [Python SDK](https://docs.cloud.coinbase.com/advanced-trade-api/docs/sdk-overview) that is pretty much the same as what you are aiming to do here.
> 
> I was just wondering how the official SDK affects the plans for this codebase. I hope to continue using this one since that's what I already started using and I prefer how this one creates objects instead of just dumping the json. So I hope the official SDK eases development of this one rather than replacing it.

Thanks for reaching out, @jonandtice. Yes, I'm aware of the new Python SDK from Coinbase. However, I'm dedicated to continuing support for this project for several reasons:
- **Migration**: My crypto projects are all integrated with this library. There isn't a compelling reason to switch to Coinbase's new SDK at this time.
- **Quality**: I appreciate our library's design, typing, and testing safeguards. In comparison, the Coinbase SDK seems rather basic at this stage.
- **Trust**: There's a history of Coinbase deprecating their previous Python SDKs, so I find more stability and reliability in our current library.
- **Bias**: Admittedly, I'm partial to our library. It was the first to support Coinbase Advanced Trade's new API, has garnered thousands of downloads, and has been in use for over a year. This familiarity gives me a greater sense of trust and confidence in its performance – and, of course, as its creator, I have a special fondness for it.

I hope this helps.

**Reply by jonandtice** [reply]
Awesome! Good to hear. I'm really wanting the portfolio functions to be implemented since I would like to get a total balance for each "account" and currently get_account() only reports available_balance and hold, ignoring any amount that is earning staking rewards (ETH is an exception, there is an "ETH2" account that is the earning amount). The amounts that would be reported by "Get Portfolio Breakdown" spot_position seem to what I'm looking for.

**Reply by KmiQ** [reply]
yup, I will be adding support in the upcoming weeks for the new Portfolios endpoints Coinbase released.

**Reply by jonandtice** [reply]
To add to the above, I manually ran List Portfolios to get my default portfolio then ran Get Portfolio Breakdown with that uuid and it listed two accounts for the coins that I am staking. So I was able to get account uuids for those and if I pass those uuids to get_account(), it returns a "Staked XXX" account name for the staking accounts. So now I'm wondering why list_accounts_all() isn't giving me the staked accounts. Should I pass this on to Coinbase or am I missing something?
